pandas/tseries/offsets.py

```diff
@@ -73,6 +78,7 @@
     "Week",
     "WeekOfMonth",
     "Easter",
+    "Tick",
```

Comment:
When I add `Tick` to `__all__`,  `pytest -k test_sub_n_gt_1_ticks  pandas/tests/arithmetic/test_period.py` fails with:

```py
cls = <class 'pandas.core.dtypes.dtypes.PeriodDtype'>, freq = <4 * Ticks>

    def __new__(cls, freq=None):
        """
        Parameters
        ----------
        freq : frequency
        """
        if isinstance(freq, PeriodDtype):
            return freq

        elif freq is None:
            # empty constructor for pickle compat
            # -10_000 corresponds to PeriodDtypeCode.UNDEFINED
            u = dtypes.PeriodDtypeBase.__new__(cls, -10_000)
            u._freq = None
            return u

        if not isinstance(freq, BaseOffset):
            freq = cls._parse_dtype_strict(freq)

        try:
            return cls._cache_dtypes[freq.freqstr]
        except KeyError:
>           dtype_code = freq._period_dtype_code
E           AttributeError: 'pandas._libs.tslibs.offsets.Tick' object has no attribute '_period_dtype_code'

pandas/core/dtypes/dtypes.py:863: AttributeError
```

I think the only runtime effect this should have is on `from pandas.tseries.offset import *` Why is `Tick` (and the other unused imports) imported in this file? I assume they are imported for some side effect?

@jbrockmendel you made changes in #34227 which might help explain this weird behavior

Reply:
This is bc Tick never gets instantiated directly, just Tick subclasses

Reply:
Thanks, it seems that when I add Tick to `__all__`, `Tick` is somehow added to `tick_classes` which causes the `Tick` classes to be tested itself.

Reply:
`tick_classes` is defined here
https://github.com/pandas-dev/pandas/blob/a853022ea154dd38dd759300ee50b456f3a9ddf6/pandas/tests/tseries/offsets/test_ticks.py#L40
but I cannot see where `Tick` is added to the list. Is there a good way to debug fixtures with pytest?

Even though `Tick` cannot/shouldn't be instantiated, I think it should be public for users to annotate functions accepting any Tick-subclass.

Reply:
found it :) the fixture is defined in https://github.com/pandas-dev/pandas/blob/a853022ea154dd38dd759300ee50b456f3a9ddf6/pandas/conftest.py#L1690